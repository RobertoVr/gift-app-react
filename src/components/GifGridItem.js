import React from 'react'

export const GifGridItem = ({ id, title, url }) => {
    return (
        <div>
            <p>{title}</p>
            <img src={url} alt="gif" />
        </div>
    )
}
