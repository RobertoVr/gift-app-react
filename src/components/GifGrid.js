import React, { useState, useEffect } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);

    useEffect(() => {
        getGifs();
    }, []);

    const getGifs = async () => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=Goku&limit=10&api_key=PVTSe8E8rAfl36W4IWV3d2GoSMEkLoVh';
        const respuesta = await fetch(url);
        const { data } = await respuesta.json();
        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })
        setImages(gifs);
    }

    return (
        <div>
            {
                images.map((img) => (
                    <GifGridItem
                        key={img.id}
                        {...img}
                    />
                ))
            }
        </div>
    )
}
