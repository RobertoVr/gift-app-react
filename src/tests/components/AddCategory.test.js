import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas eb <AddCategory />', () => {

    const setCategories = () => { };
    const wrapper = shallow(<AddCategory setCategories={setCategories} />)

    test('Debe de mostrar correctamente el componente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    test('Debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'hola mundo';
        input.simulate('change', { target: { value } });
        const p = wrapper.find('p').text();
        expect(p.trim()).toBe(value);
    });


});
