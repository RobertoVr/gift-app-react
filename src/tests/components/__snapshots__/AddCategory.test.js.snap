// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas eb <AddCategory /> Debe de mostrar correctamente el componente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
